File: GomokuAIJavaProject/AIGomoku/Position.java
package AIGomoku;
import java.util.Objects;

public class Position {
	private int x;
	private int y;
	
	public Position(int y, int x) {
		super();
		this.x = x;
		this.y = y;
	}	
	
	public int getX() {
		return x;
	}

	public int getY() {
		return y;
	}

	public boolean isNear(Position pos) {
		
		int posy = pos.getY();
		int posx = pos.getX();
		if((Math.abs(posx - x) == 1 || Math.abs(posx - x) == 0) && (Math.abs(posy - x) == 1 || Math.abs(posy - y) == 0)) {
			return true;
		}
		
		return false;
	}

	@Override
	public int hashCode() {
		return Objects.hash(x, y);
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Position other = (Position) obj;
		return x == other.x && y == other.y;
	}

	@Override
	public String toString() {
		return "Position [y=" + (y+1) + ", x=" + (x+1) + "]";
	}
	
	
	
	
	
}

File: CPP/src/Position.cpp
#include "Position.hpp"

Position::Position(unsigned int x, unsigned int y)
{
    this->x = x;
    this->y = y;
}

Position::Position(const Position &position) 
{
    this->x = position.x;
    this->y = position.y;
}

unsigned int Position::getX() const
{
    return this->x;
}

unsigned int Position::getY() const
{
    return this->y;
}

void Position::setY(unsigned int y)
{
    this->y = y;
}

void Position::setX(unsigned int x)
{
    this->x = x;
}

bool Position::isNear(Position* pos)
{
    unsigned int posy = pos->getY();
    unsigned int posx = pos->getX();
    if((std::abs((int)(posx - x)) == 1 || std::abs((int)(posx - x)) == 0) && (std::abs((int)(posy - x)) == 1 || std::abs((int)(posy - y)) == 0)) {
        return true;
    }
    return false;
}

bool Position::operator==(const Position& p1) const
{
    if(this->x == p1.x && p1.y == this->y){
        return true;
    }
    return false;
}